juju/utils.py

```diff
@@ -457,27 +458,30 @@ def get_base_from_origin_or_channel(origin_or_channel, series=None):
     return client.Base(channel=channel, name=os_name)
 
 
-def series_for_charm(requested_series, supported_series):
+def series_for_charm(requested_series: str, supported_series: Sequence[str]):
     """series_for_charm takes a requested series and a list of series supported by a
     charm and returns the series which is relevant.
     If the requested series is empty, then the first supported series is used,
     otherwise the requested series is validated against the supported series.
+
+    supported_series follows Juju API convention (best first):
+    - LTS in reverse chronological order
+    - then regular releases in reverse chronological order
     """
     if len(supported_series) == 1 and supported_series[0] == '':
         raise JujuError("invalid supported series reported by charm : ['']")
-    if len(supported_series) == 0:
-        if requested_series == '':
-            raise JujuError("missing series")
+    if not supported_series and requested_series:
         return requested_series
+    elif not requested_series:
+        raise JujuError("missing series")
+    elif not requested_series:
+        # use the charm default
+        return supported_series[0]
 
-    # use the charm default
-    if requested_series == '':
-        return supported_series[-1]
```

Comment:
This was incorrect. It should have been 0. 

https://github.com/juju/charm/blob/bd9cd4649313dc47c456713e33543c957ce1eaad/charm.go#L126-L129